src/modules/hooks/connection_hooks.js

```diff
@@ -0,0 +1,54 @@
+const BaseHooks = require('./base_hooks');
+
+class ConnectionHooks extends BaseHooks {
+  constructor(hooks) {
+    super();
+
+    if (!hooks) {
+      return;
+    }
+    if (hooks.afterConnect) {
+      this.afterConnect(hooks.afterConnect);
+    }
+    if (hooks.beforeConnect) {
+      this.beforeConnect(hooks.beforeConnect);
+    }
+  }
+
+  /**
+   * Registers callback/callbacks to be invoked before creating a connection to rabbitmq.
+   * The callback is invoked with 'this' set to the connection wrapper instance creating the connection, and a single "payload" argument of the following shape:
```

Comment:
Do we really want "this" to be binded to our internal connection object? it means we are exposing the internals of the library and this can misused.

Reply:
It is the same object that is exposed when you `require('armavmq')({...}).connection`